### DISCUSSION TITLE
How to check average clustering diameter of clusters formed by ClusterData()

### DISCUSSION THREAD

**Discussion by meenamohit** [pyg-team/pytorch_geometric#9232] [General]
I want to know the average clustering diameter of clusters formed using metis (ClusterData) function. how can I check that?

I tried converting the graph to networkx and then applied diameter function but it says that graph is not strongly connected

**Comment by rusty1s** [comment]
That would be my preferred solution as well. I think it is expected that `ClusterData` will not necessarily return a strongly connected graph.

**Comment by meenamohit** [comment]
Thanks for the reply @rusty1s, Do you think there's any alternate approach to do so?

**Reply by rusty1s** [reply]
How about finding the connected components and then apply the diameter function to all of them? This would be similar to our `LargestConnectedComponents` transform.

**Comment by meenamohit** [comment]
I've tried that too but it throws the same error

**Reply by rusty1s** [reply]
Do you mind sharing a small example to reproduce?
